src/tigerbeetle-java/src/main/java/com/tigerbeetle/Account.java

```diff
@@ -50,99 +56,208 @@ public Account() {
+     *
+     * @return An {@link java.util.UUID} representing an integer-encoded UUIDv4 or any other unique
+     *         128-bit integer.
+     */
+    public UUID getUserData() {
+        return userData;
+    }
+
+    /**
+     * Sets the secondary identifier to link this account to an external entity.
+     * <p>
+     * May be zero, null values are converted to zero.
+     *
+     * @param userData An {@link java.util.UUID} representing an integer-encoded UUIDv4 or any other
+     *        unique 128-bit integer.
+     */
+    public void setUserData(UUID userData) {
+        if (userData == null) {
+            this.userData = ZERO;
+        } else {
+            this.userData = userData;
+        }
+    }
+
+    /**
+     * Gets an identifier used to enforce transfers between the same ledger.
+     *
+     * @return A 32-bit integer.
+     */
+    public int getLedger() {
+        return ledger;
+    }
+
+    /**
+     * Sets an identifier used to enforce transfers between the same ledger.
+     * <p>
+     * Must be non-zero.
+     * <p>
+     * Example: 1 for USD and 2 for EUR.
+     *
+     * @param ledger A 32-bit integer defined by the user.
+     */
+    public void setLedger(int ledger) {
+        this.ledger = ledger;
+    }
+
+    /**
+     * Gets a reason for the transfer.
+     *
+     * @return A 16-bits unsigned integer.
+     */
+    public int getCode() {
         return code;
     }
 
+    /**
+     * Sets a reason for the transfer.
+     * <p>
+     * Must be non-zero.
+     * <p>
+     * Example: 1 for deposit, 2 for settlement.
+     *
+     * @param code A 16-bits unsigned integer defined by the user.
+     * @throws IllegalArgumentException if code is negative or greater than 65535.
+     */
     public void setCode(int code) {
         if (code < 0 || code > Character.MAX_VALUE)
-            throw new IllegalArgumentException("Code must be a unsigned 16 bits value");
+            throw new IllegalArgumentException("Code must be a 16-bits unsigned integer");
 
         this.code = (short) code;
     }
 
-    public short getFlags() {
+    /**
+     * Gets the behavior during transfers.
+     *
+     * @return A 16-bits unsigned integer bit mask.
+     * @see com.tigerbeetle.AccountFlags
+     */
+    public int getFlags() {
         return flags;
     }
 
+    /**
+     * Sets the behavior during transfers.
+     * <p>
+     * Example: {@code AccountFlags.DEBITS_MUST_NOT_EXCEED_CREDITS} to force debts not to exceed
+     * credits.
+     *
+     * @see com.tigerbeetle.AccountFlags
+     * @param flags A 16-bits unsigned integer bit mask.
+     * @throws IllegalArgumentException if flags is negative or greater than 65535.
+     */
     public void setFlags(int flags) {
         if (flags < 0 || flags > Character.MAX_VALUE)
-            throw new IllegalArgumentException("Flags must be a unsigned 16 bits value");
+            throw new IllegalArgumentException("Flags must be a 16-bits unsigned integer");
```

Comment:
I'd say `16-bit unsigned integer` not `16-bits unsigned integer`. I can't explain it but the former seems more natural

Everywhere you say X-bits unsigned integer it should probably be X-bit unsigned integer